project/MimaExcludes.scala

```diff
@@ -465,7 +465,15 @@ object MimaExcludes {
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.deploy.SparkHadoopUtil.appendS3AndSparkHadoopConfigurations"),
 
     // [SPARK-29348] Add observable metrics.
-    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.streaming.StreamingQueryProgress.this")
+    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.streaming.StreamingQueryProgress.this"),
+
+    // [SPARK-30419][ML] Make IsotonicRegression extend Regressor
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.fit"),
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.setFeaturesCol"),
```

Comment:
Sorry for the dumb questions, but, hm, this removes several key setters. Is the API that different, or, how do you set the feature col, etc now?

Reply:
I was confused as well when I first saw the Mima errors lol

Here are the Mima errors:
```
[error]  * method fit(org.apache.spark.sql.Dataset)org.apache.spark.ml.regression.IsotonicRegressionModel in class org.apache.spark.ml.regression.IsotonicRegression has a different result type in current version, where it is org.apache.spark.ml.Model rather than org.apache.spark.ml.regression.IsotonicRegressionModel
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.fit")
[error]  * method setFeaturesCol(java.lang.String)org.apache.spark.ml.regression.IsotonicRegression in class org.apache.spark.ml.regression.IsotonicRegression has a different result type in current version, where it is org.apache.spark.ml.Predictor rather than org.apache.spark.ml.regression.IsotonicRegression
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.setFeaturesCol")
[error]  * method setLabelCol(java.lang.String)org.apache.spark.ml.regression.IsotonicRegression in class org.apache.spark.ml.regression.IsotonicRegression has a different result type in current version, where it is org.apache.spark.ml.Predictor rather than org.apache.spark.ml.regression.IsotonicRegression
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.setLabelCol")
[error]  * method setPredictionCol(java.lang.String)org.apache.spark.ml.regression.IsotonicRegression in class org.apache.spark.ml.regression.IsotonicRegression has a different result type in current version, where it is org.apache.spark.ml.Predictor rather than org.apache.spark.ml.regression.IsotonicRegression
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegression.setPredictionCol")
[error]  * method setFeaturesCol(java.lang.String)org.apache.spark.ml.regression.IsotonicRegressionModel in class org.apache.spark.ml.regression.IsotonicRegressionModel has a different result type in current version, where it is org.apache.spark.ml.PredictionModel rather than org.apache.spark.ml.regression.IsotonicRegressionModel
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegressionModel.setFeaturesCol")
[error]  * method setPredictionCol(java.lang.String)org.apache.spark.ml.regression.IsotonicRegressionModel in class org.apache.spark.ml.regression.IsotonicRegressionModel has a different result type in current version, where it is org.apache.spark.ml.PredictionModel rather than org.apache.spark.ml.regression.IsotonicRegressionModel
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.ml.regression.IsotonicRegressionModel.setPredictionCol")
```
The APIs are still the same, but the return types are different. Before the change, setXXX is in ```IsotonicRegression``` and the return type is ```IsotonicRegression```

```
  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
```
After the change, setXXX is in the super class ```Predictor``` and the return type is ```Predictor```
```
  def setFeaturesCol(value: String): Learner = set(featuresCol, value).asInstanceOf[Learner]
```

Reply:
Hm, weird, because Learner is the type IsotonicRegression here. So it shouldn't be a real change. I wonder if it's just a MiMa problem. 

So as far as you know this isn't changing any APIs right?